### PR TITLE
Add cleanup parameter.

### PR DESCRIPTION
**Pull request by chouseknecht**
##### ISSUE TYPE
- Feature Pull Request
##### COMPONENT NAME

docker_container.py
##### ANSIBLE VERSION

```
ansible 2.2.0 (devel be66ebca37) last updated 2016/07/09 01:55:50 (GMT -400)
  lib/ansible/modules/core: (fix_4088 5b441b210b) last updated 2016/07/09 04:55:44 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD 618e740d52) last updated 2016/07/08 17:30:59 (GMT -400)
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### SUMMARY

Per request 4088 add a method for removing a container as soon as it finishes running.

### CODE REVIEW

**Comment by gregdek** [comment]
maintainer unknown ping @gregdek @resmo @robynbergeron.

**Comment by gregdek** [comment]
Thanks @chouseknecht for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by PikachuEXE** [comment]
The behaviour looks good to me
But the state name `cleaned` sounds strange to me
Just like you won't run a container with state `stopped` or `removed`

I think it should be something that describe the nature of the behaviour:
- "Started"
- "Will be cleanup"

`started_with_cleanup`? << Seems terrible
`started_onetime`? << So strange
`running_once`? << Not very good but I can't come up with better

**Comment by bcoca** [comment]
A simple `cleanup=no|yes` sounds better.

I would even separate started/stopped into `started=yes|no`

**Comment by chouseknecht** [comment]
Modified to use `cleanup=no|yes`.

File: cloud/docker/docker_container.py [hunk 1]
```diff
@@ -40,6 +40,11 @@
       - List of capabilities to add to the container.
     default: null
     required: false
+  cleanup:
+    description:
+      - Use with I(detach) to remove the container after successful execution.
+    default: false
+    required: false
```
Comment (bcoca) [inline cloud/docker/docker_container.py:47]: missing version_added (2.2)

**Pull request merged**